File: examples/minimal-ts/agent.ts
import { SelfAgent } from "@selfxyz/agent-sdk";

// Create an agent from a private key
const agent = new SelfAgent({
  privateKey: process.env.AGENT_PRIVATE_KEY!,
  network: "testnet",
});

// Check registration status
const registered = await agent.isRegistered();
console.log("Registered:", registered);

if (registered) {
  const info = await agent.getInfo();
  console.log("Agent ID:", info.agentId);
  console.log("Verified:", info.isVerified);

  // Make a signed request to a protected API
  const res = await agent.fetch("http://localhost:3000/api/data", {
    method: "POST",
    body: JSON.stringify({ message: "Hello from a verified agent" }),
  });
  console.log("Response:", res.status, await res.text());
}

File: examples/minimal-python/agent.py
# SPDX-License-Identifier: MIT

"""Minimal agent: sign and send a verified request."""
import os
from self_agent_sdk import SelfAgent

agent = SelfAgent(
    private_key=os.environ["AGENT_PRIVATE_KEY"],
    network="testnet",
)

# Check registration
print("Registered:", agent.is_registered())

if agent.is_registered():
    info = agent.get_info()
    print(f"Agent ID: {info.agent_id}, Verified: {info.is_verified}")

    # Send a signed request
    res = agent.fetch(
        "http://localhost:8000/api/data",
        method="POST",
        body='{"message": "Hello from a verified agent"}',
    )
    print(f"Response: {res.status_code} {res.text}")
